### PR TITLE
Implements parsing for #pragma, handles the multiple case niavely

### PR DESCRIPTION
**Pull request by LemonInTheDark**
I'm using a second hashmap to track multiple'd files. Not sure that this is the best way of going about it, but it's a living

### CODE REVIEW

File: crates/dreammaker/src/preprocessor.rs [hunk 1]
```diff
@@ -952,6 +967,18 @@ impl<'ctx> Preprocessor<'ctx> {
                         expect_token!((text) = Token::String(text));
                         self.context.register_error(DMError::new(self.last_input_loc, format!("#{} {}", ident, text)));
                     }
+                    "pragma" if disabled => {}
+                    "pragma" => {
+                        expect_token!((text) = Token::String(text));
```
Comment (SpaceManiac) [inline crates/dreammaker/src/preprocessor.rs:972]: You probably mean Ident here, not String

File: crates/dreammaker/src/preprocessor.rs [hunk 2]
```diff
@@ -952,6 +967,18 @@ impl<'ctx> Preprocessor<'ctx> {
...
+                        match text.as_str() {
+                            "multiple" => {
+                                if let Some(file_id) = self.context.get_file(self.env_file.as_path()) {
```
Comment (SpaceManiac) [inline crates/dreammaker/src/preprocessor.rs:975]: `self.env_file` is the root, not the file currently being processed

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: crates/dreammaker/src/preprocessor.rs (hunk 1 above)
Comment (LemonInTheDark) [reply crates/dreammaker/src/preprocessor.rs:972]: I changed it, but are you sure? #warn uses String() and its input isn't "'d at all, it's just the text after the #warn

Comment (SpaceManiac) [reply crates/dreammaker/src/preprocessor.rs]: It's stupid, but there's a hack in lexer.rs for this: `if ident == "warn" || ident == "error" {`

Comment (LemonInTheDark) [reply crates/dreammaker/src/preprocessor.rs]: in fucking tears

Comment (LemonInTheDark) [reply crates/dreammaker/src/preprocessor.rs]: should I add to that or just keep as is

Comment (SpaceManiac) [reply crates/dreammaker/src/preprocessor.rs]: As written is fine. The reason is that #warn and #error want to treat the whole rest of the line as one big string. #pragma is (for now...) followed by ordinary tokens.

Comment (LemonInTheDark) [reply crates/dreammaker/src/preprocessor.rs]: alright. is there anything else you want me to do here?

**Pull request merged**